### PR TITLE
Add max value in flight flag

### PR DESCRIPTION
**Pull request by vctt94**
This PR adds an optional max value in flight flag to openchannel.

Starts #800. As it is my first PR, I am adding only one flag to see if I am going the right direction, then on upcoming PRs I can add more flags on the same PR.

### CODE REVIEW

File: cmd/lncli/commands.go [hunk 1]
```diff
@@ -446,6 +446,11 @@ var openChannelCommand = cli.Command{
 				"not set, we will scale the value according to the " +
 				"channel size",
 		},
+		cli.Int64Flag{
+			Name: "max_value_in_flight",
+			Usage: "(optional) the maximum amount of coins that can be " +
```
[the remark this answers is not in the document]
Comment (halseth) [reply cmd/lncli/commands.go]: No, I mean `btc`/`sat`/`msat`

File: server.go [hunk 2]
```diff
@@ -1866,7 +1867,7 @@ func (s *server) DisconnectPeer(pubKey *btcec.PublicKey) error {
 func (s *server) OpenChannel(nodeKey *btcec.PublicKey,
 	localAmt btcutil.Amount, pushAmt, minHtlc lnwire.MilliSatoshi,
 	fundingFeePerVSize lnwallet.SatPerVByte, private bool,
-	remoteCsvDelay uint16) (chan *lnrpc.OpenStatusUpdate, chan error) {
+	remoteCsvDelay uint16, maxValueInFlight lnwire.MilliSatoshi) (chan *lnrpc.OpenStatusUpdate, chan error) {
```
[the remark this answers is not in the document]
Comment (vctt94) [reply server.go]: fixed

File: server.go [hunk 3]
```diff
@@ -1860,14 +1860,32 @@ func (s *server) DisconnectPeer(pubKey *btcec.PublicKey) error {
 	return nil
 }
 
+// OpenChannelStruct is respresenting the possibles flags used when opening a channel.
+type openChannelStruct struct {
```
[the remark this answers is not in the document]
Comment (vctt94) [reply server.go]: fixed

**Comment by vctt94** [comment]
Thanks for reviewing my PR again, fixed your last issues.

File: cmd/lncli/commands.go [hunk 4]
```diff
@@ -446,6 +446,11 @@ var openChannelCommand = cli.Command{
 				"not set, we will scale the value according to the " +
 				"channel size",
 		},
+		cli.Int64Flag{
+			Name: "max_value_in_flight",
```
Comment (halseth) [inline cmd/lncli/commands.go]: make this `max_value_in_flight_msat`

File: fundingmanager_test.go [hunk 5]
```diff
@@ -2126,6 +2164,12 @@ func TestFundingManagerCustomChannelParameters(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// The max value in flight Alice can have should be maxValueAcceptChannel, which is
+	// the default value and the maxium Bob can offer should be maxValueInFlight.
+	if err := assertMaxValueInFlight(resCtx, maxValueAcceptChannel, maxValueInFlight); err != nil {
```
Comment (halseth) [inline fundingmanager_test.go]: nit: long line

File: fundingmanager_test.go [hunk 6]
```diff
@@ -2139,4 +2183,8 @@ func TestFundingManagerCustomChannelParameters(t *testing.T) {
 	if err := assertMinHtlc(resCtx, minHtlc, 5); err != nil {
 		t.Fatal(err)
 	}
+
+	if err := assertMaxValueInFlight(resCtx, maxValueInFlight, maxValueAcceptChannel); err != nil {
```
Comment (halseth) [inline fundingmanager_test.go]: nit: long line

File: lnrpc/rpc.proto [hunk 7]
```diff
@@ -990,6 +990,9 @@ message OpenChannelRequest {
 
     /// The delay we require on the remote's commitment transaction. If this is not set, it will be scaled automatically with the channel size.
     uint32 remote_csv_delay = 10 [json_name = "remote_csv_delay"];
+
+    /// The maximum amount of coins in millisatoshi that can be pending within the channel.
+	int64 MaxValueInFlight = 11 [json_name = "max_value_in_flight_msat"];
```
[the remark this answers is not in the document]
Comment (vctt94) [reply lnrpc/rpc.proto]: tab indented. Thanks for noticing

File: lnrpc/rpc.proto [hunk 8]
```diff
@@ -992,7 +992,7 @@ message OpenChannelRequest {
     uint32 remote_csv_delay = 10 [json_name = "remote_csv_delay"];
 
     /// The maximum amount of coins in millisatoshi that can be pending within the channel.
-	int64 MaxValueInFlight = 11 [json_name = "max_value_in_flight_msat"];
+    int64 MaxValueInFlight = 11 [json_name = "max_value_in_flight_msat"];
```
Comment (halseth) [inline lnrpc/rpc.proto]: This should be a fixup to the previous `lnrpc` commit :)

Comment (vctt94) [reply lnrpc/rpc.proto]: it is in the right commit now :)

**Comment by vctt94** [comment]
rebased

**Comment by halseth** [comment]
@vctt94 Hey, this needs a rebase due to recent changes 😄

**Comment by vctt94** [comment]
rebased

File: fundingmanager.go [hunk 9]
```diff
@@ -2559,7 +2558,6 @@ func (f *fundingManager) handleInitFundingMsg(msg *initFundingMsg) {
 		msg.err <- err
 		return
 	}
-	addMaxValueFlag
```
Comment (halseth) [inline fundingmanager.go]: this belongs to the fundingmanager commit

Comment (vctt94) [reply fundingmanager.go]: Sorry about that. 

fixed.

File: server.go [hunk 10]
```diff
@@ -2074,14 +2075,33 @@ func (s *server) DisconnectPeer(pubKey *btcec.PublicKey) error {
 	return nil
 }
 
+// openChannelReq is respresenting the possibles flags used
```
Comment (halseth) [inline server.go]: nit: possible

File: server.go [hunk 11]
```diff
@@ -2074,14 +2075,33 @@ func (s *server) DisconnectPeer(pubKey *btcec.PublicKey) error {
...
 // OpenChannel sends a request to the server to open a channel to the specified
 // peer identified by nodeKey with the passed channel funding parameters.
 //
 // NOTE: This function is safe for concurrent access.
-func (s *server) OpenChannel(nodeKey *btcec.PublicKey,
-	localAmt btcutil.Amount, pushAmt, minHtlc lnwire.MilliSatoshi,
-	fundingFeePerVSize lnwallet.SatPerVByte, private bool,
-	remoteCsvDelay uint16) (chan *lnrpc.OpenStatusUpdate, chan error) {
+func (s *server) OpenChannel(openChannel *openChannelReq) (chan *lnrpc.OpenStatusUpdate, chan error) {
+
```
Comment (halseth) [inline server.go]: nit: unnecessary newline

File: server.go [hunk 12]
```diff
@@ -2074,14 +2075,33 @@ func (s *server) DisconnectPeer(pubKey *btcec.PublicKey) error {
...
 // OpenChannel sends a request to the server to open a channel to the specified
 // peer identified by nodeKey with the passed channel funding parameters.
 //
 // NOTE: This function is safe for concurrent access.
-func (s *server) OpenChannel(nodeKey *btcec.PublicKey,
-	localAmt btcutil.Amount, pushAmt, minHtlc lnwire.MilliSatoshi,
-	fundingFeePerVSize lnwallet.SatPerVByte, private bool,
-	remoteCsvDelay uint16) (chan *lnrpc.OpenStatusUpdate, chan error) {
+func (s *server) OpenChannel(openChannel *openChannelReq) (chan *lnrpc.OpenStatusUpdate, chan error) {
```
Comment (halseth) [inline server.go]: line too long

**Comment by vctt94** [comment]
fixed

**Comment by vctt94** [comment]
rebased

**Comment by vctt94** [comment]
Travis is failing and I am not sure why. Can someone help me?

**Comment by halseth** [comment]
Did you rebase on master? It should have been fixed there

**Comment by vctt94** [comment]
I thought I did, but I was not on latest master, thanks for your help.

Now the build passes but I have a different error. Should I add more tests? If that is the case, where can I add those?

**Comment by halseth** [comment]
@vctt94 Don't worry about it for now 😄  We are rewriting parts of `fundingmanager`, which will require some changes to this PR. I will let you know when it is ready to rebase it!

**Comment by vctt94** [comment]
okay, thanks :)

**Comment by guggero** [comment]
Seems this PR has been waiting for changes in `fundingmanager`. @halseth have these landed yet?

**Comment by halseth** [comment]
Yes, they have landed :)

**Comment by vctt94** [comment]
rebased :)

File: cmd/lncli/commands.go [hunk 13]
```diff
@@ -659,6 +659,11 @@ var openChannelCommand = cli.Command{
 				"transaction must satisfy",
 			Value: 1,
 		},
+		cli.Int64Flag{
+			Name: "max_value_in_flight_msat",
```
Comment (cfromknecht) [inline cmd/lncli/commands.go]: should we prefix with `remote_` to make it clear? not always clear to people that you tell the other party what values to use

File: lnrpc/rpc.proto [hunk 14]
```diff
@@ -1606,6 +1606,8 @@ message OpenChannelRequest {
 
     /// Whether unconfirmed outputs should be used as inputs for the funding transaction.
     bool spend_unconfirmed = 12 [json_name = "spend_unconfirmed"];
+    /// The maximum amount of coins in millisatoshi that can be pending within the channel.
+    int64 MaxValueInFlight = 13 [json_name = "max_value_in_flight_msat"];
```
Comment (cfromknecht) [inline lnrpc/rpc.proto]: this commit should be swapped with the prior, that way we don't get build errors when trying to bisect :)

**Comment by vctt94** [comment]
Rebased

**Comment by cfromknecht** [comment]
@vctt94 ready for another rebase? I'd like to get this in early in the 0.11 cycle so that we can get you out of rebase hell :)

**Comment by vctt94** [comment]
Rebased

File: server.go [hunk 15]
```diff
@@ -3159,6 +3159,8 @@ type openChanReq struct {
 	// This value is optional, so may be nil.
 	shutdownScript lnwire.DeliveryAddress
 
+	maxValueInFlight lnwire.MilliSatoshi
```
Comment (wpaulino) [inline server.go:3162]: Nit: missing godoc.

File: cmd/lncli/cmd_open_channel.go [hunk 16]
```diff
@@ -163,6 +163,11 @@ var openChannelCommand = cli.Command{
 				"as a base and add the new channel output to " +
 				"it instead of creating a new, empty one.",
 		},
+		cli.Int64Flag{
```
Comment (wpaulino) [inline cmd/lncli/cmd_open_channel.go:166]:
```suggestion
		cli.Uint64Flag{
```

File: lnrpc/rpc.proto [hunk 17]
```diff
@@ -1704,6 +1704,9 @@ message OpenChannelRequest {
     carried out in an interactive manner (PSBT based).
     */
     FundingShim funding_shim = 14;
+
+    /// The maximum amount of coins in millisatoshi that can be pending within the channel.
```
[the remark this answers is not in the document]
Comment (wpaulino) [reply lnrpc/rpc.proto:1708]: Note that this only applies to the remote party.

File: cmd/lncli/cmd_open_channel.go [hunk 18]
```diff
@@ -191,6 +196,7 @@ func openChannel(ctx *cli.Context) error {
 		MinConfs:         minConfs,
 		SpendUnconfirmed: minConfs == 0,
 		CloseAddress:     ctx.String("close_address"),
+		MaxValueInFlight: ctx.Int64("remote_max_value_in_flight_msat"),
```
Comment (wpaulino) [inline cmd/lncli/cmd_open_channel.go:199]: We strive to have each commit compile independently, so the RPC changes should come before this.

File: lnrpc/rpc.proto [hunk 19]
```diff
@@ -1704,6 +1704,9 @@ message OpenChannelRequest {
...
+    int64 MaxValueInFlight = 15 [json_name = "max_value_in_flight_msat"];
```
Comment (wpaulino) [inline lnrpc/rpc.proto:1709]: I haven't confirmed, but you may need to update your protobuf generation dependencies if you haven't already as we've updated them recently (see https://github.com/lightningnetwork/lnd/tree/master/lnrpc#generate-protobuf-definitions).

**Comment by vctt94** [comment]
Thanks for the review @wpaulino. I addressed your comments.

File: cmd/lncli/cmd_open_channel.go [hunk 20]
```diff
@@ -191,6 +196,7 @@ func openChannel(ctx *cli.Context) error {
 		MinConfs:         minConfs,
 		SpendUnconfirmed: minConfs == 0,
 		CloseAddress:     ctx.String("close_address"),
+		MaxValueInFlight: ctx.Uint64("remote_max_value_in_flight_msat"),
```
Comment (wpaulino) [inline cmd/lncli/cmd_open_channel.go:199]: The `lnrpc` commit (9842639e03b3e510dcf385c92c3b28d5b8121ffb) should come before this one (56dd35e3424d7ae5da512390febe1720cb524a34).

File: lnrpc/rpc.proto [hunk 21]
```diff
@@ -1558,6 +1558,12 @@ message OpenChannelRequest {
     carried out in an interactive manner (PSBT based).
     */
     FundingShim funding_shim = 14;
+
+    /*
+    The maximum amount of coins in millisatoshi that can be pending within
+    the channel. It only applies to the remote party.
+    */
+    uint64 max_value_in_flight = 15;
```
Comment (wpaulino) [inline lnrpc/rpc.proto:1566]:
```suggestion
    uint64 remote_max_value_in_flight = 15;
```

To keep it consistent with `remove_csv_delay`, etc.

**Comment by vctt94** [comment]
Okay, I believe I cherry-picked it properly now, but tests still failiing

File: lnrpc/rpc.proto (hunk 21 above)
Comment (wpaulino) [inline lnrpc/rpc.proto:1566]: Could you rename this to `remote_max_value_in_flight_msat`?

**Comment by wpaulino** [comment]
Build is broken after the latest push.

**Comment by vctt94** [comment]
my bad. Fixed it

**Comment by cfromknecht** [comment]
cc @wpaulino. I would like to merge this before #4157 to prevent @vctt94 from having to rebase again

**Pull request merged**